kbase-extension/static/kbase/js/widgets/narrative_core/kbaseDataCard.js

```diff
@@ -0,0 +1,209 @@
+                        e.stopPropagation();
+                        window.open('/#people/' + object_info[5]);
+                    }.bind( null, object_info));
+                $subcontent.append($byUser);
+            }
+
+            //tooltip for long title
+            if (isShortened) {
+                $name.tooltip({
+                    title: entry.name,
+                    placement: 'bottom',
+                    delay: {
+                        show: Config.get('tooltip').showDelay,
+                        hide: Config.get('tooltip').hideDelay
+                    }
+                });
+            }
+            
+            //create card
+            var actionButtonClick = function (e) {
+                if(!entry.ws_name){
+                    return;
+                }
+                e.preventPropagation; 
+                var updateButton = function () {
+                    var thisBtn = $(this).children()[0];
+                    var thisHolder = this;
+                    $(this).html('<img src="' + self.options.loadingImage + '">');
+                    Promise.resolve(self.serviceClient.sync_call(
+                        'NarrativeService.copy_object',
+                        [{
+                            ref: object_info[6] + '/' + object_info[0],
+                            target_ws_name: entry.ws_name,
+                        }]
+                    ))
+                        .then(function () {
+                            $(thisHolder).html('');
+                            $(thisBtn).text(' Copy');
+                            $(thisHolder).append(thisBtn);
+                            self.trigger('updateDataList.Narrative');
+                        })
+                        .catch(function (error) {
+                            $(thisBtn).html('Error');
+                            if (error.error && error.error.message) {
+                                if (error.error.message.indexOf('may not write to workspace') >= 0) {
+                                    self.options.$importStatus.html($('<div>').css({ 'color': '#F44336', 'width': '500px' }).append('Error: you do not have permission to add data to this Narrative.'));
+                                } else {
+                                    self.options.$importStatus.html($('<div>').css({ 'color': '#F44336', 'width': '500px' }).append('Error: ' + error.error.message));
+                                }
+                            } else {
+                                self.options.$importStatus.html($('<div>').css({ 'color': '#F44336', 'width': '500px' }).append('Unknown error!'));
+                            }
+                            console.error(error);
+                        });
+                };
+                if ($(this).text().split(' ')[1] === 'Copy') {
+                    var dialog = new BootstrapDialog({
+                        title: 'Item already exists in workspace under same name.',
```

Comment:
We don't really expose the "workspace" concept to users. Instead, use "Narrative". Like "An item with this name already exists in this Narrative."

Reply:
im stealing this line